### PR TITLE
Update `Gemfile` docs' link targets for `:source`

### PR DESCRIPTION
**Pull request by myersjustinc**
These two links don't go anywhere in the browser, most likely because 01fc34bebd128d758733261f16d9e9a8523d81d4 changed the text of those (and other) headers.

As far as I know (based on a quick search with `/\w+-)/`), these are the only two links with this problem.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 1.14.6 (unreleased)
+
+Documentation:
+
+  - Fix links about `:source` in `Gemfile` man page
```
Comment (segiddins) [inline CHANGELOG.md]: we don't document changes in the changelog until the release is made

Comment (myersjustinc) [inline CHANGELOG.md]: Ah, thanks for clarifying. Saw a mention of it in [the pull request guidelines](https://github.com/bundler/bundler/blob/master/doc/development/PULL_REQUESTS.md#changelogmd), so I went ahead and added it.

Should I remove that from this branch? And if so, would you prefer for me to do that in another commit or to amend my existing commit?

Comment (olleolleolle) [inline CHANGELOG.md]: (Amend the commit.)

Comment (myersjustinc) [inline CHANGELOG.md]: Amended!

**Comment by segiddins** [comment]
Thanks so much!
@bundlerbot r+
